### ISSUE TITLE
Allow for jwt builtin error handling

### ISSUE THREAD

**Issue by klarose** [open-policy-agent/opa#2765]
## Expected Behavior

When a builtin function encounters an error, it would be convenient if there were a method of treating that error as equivalent
to a rule failing, as opposed to having it terminate evaluation entirely.

## Actual Behavior

Evaluation of a rule fails entirely. There is no way to recover.

This is problematic in some cases. For example, when doing http bearer authorization I may support two cases: one where
the token is a jwt, and one where it's an arbitrary string. The jwt authorization raises an evaluation error if I pass in an arbitrary string. The only way I can see to make this work is to prevent evaluation of the bearer token as a JWT if it isn't a well-formed JWT. This means I need to repeat much of the work done already by the jwt builtins: I'd need to split the token, base64 decode it, and so on

## Steps to Reproduce the Problem

Run the following script:

```
TOKEN := "foo"

eval_as_jwt {
  [header, payload, _] := io.jwt.decode(TOKEN)
  payload["iss"] == "my-issuer"
}

eval_as_str {
  TOKEN == "foo"
}

default allowed = false

allowed {
  # token case
  eval_as_jwt
}

allowed {
  # string case
  eval_as_str
}
```

Then evaluate `allowed`:

```
> allowed
1 error occurred: io.jwt.decode(TOKEN): eval_builtin_error: io.jwt.decode: encoded JWT had no period separators
```

Allowed should be true.

## Additional Info

I'm not sure what the best way to solve this is. One option is make specific functions optionally not fail this way when operating. 

My specific issue is with jwt validation, so I'm fine with a solution targeting it explicitly: maybe we just make the verify functions return false if there is a problem with evaluation, rather than returning an error, or,  maybe we need an "is_valid_jwt" builtin which can be used to prevent further processing.

**Comment by patrick-east** [comment]
There is a draft PR for this type of thing (its a common problem with _all_ of the OPA builtins, although it affects the JWT builtins pretty often) https://github.com/open-policy-agent/opa/pull/2717 with an issue tracking it #2086 

I think in the short term there is a precedent set for having `XXX.is_valid(..)` builtins (eg `semver.is_valid(str)` and `regex.is_valid(str)`) for this kind of problem. I could see having a similar `io.jwt.is_valid(str)` which gives back a yes/no answer to whether the string is a JWT... the naming is maybe a little bit unfortunate but as long as we are clear about it in the docs it is probably fine.

**Comment by klarose** [comment]
@patrick-east Thanks! I'm glad to see that there's a generic solution in the works. It's not super urgent for me that we resolve the specific jwt case, so if it won't be too long before the generic solution is in place, I'm fine waiting.

**Comment by anderseknert** [comment]
This can be done with a Rego helper function like the one below, which should be enough for pretty much all practical purposes:

```rego
jwt_format_is_valid(token) {
    s := split(token, ".")
    count(s) == 3
    startswith(s[0], "ey")
    # Remove last line if you need to support JWE's, though I don't think OPA currently supports them anyway
    startswith(s[1], "ey")
}
```

So while possibly a nice addition to the built-ins it should be pretty easy to work around in the meantime. If added to the built-ins I think the name should be something like the above, including "format" or something else that would make it obvious that it's not validating the JWT itself but merely the structure.

**Comment by tsandall** [comment]
Since the built-in function error handling changes landed in v0.25.0 this should be solved.
